modules/opentelemetry/src/main/java/com/spotify/ffwd/opentelemetry/OpenTelemetryPluginSink.java

```diff
@@ -0,0 +1,174 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.ffwd.opentelemetry;
+
+import com.google.inject.Inject;
+import com.spotify.ffwd.model.v2.Batch;
+import com.spotify.ffwd.model.v2.Metric;
+import com.spotify.ffwd.output.PluginSink;
+import eu.toolchain.async.AsyncFramework;
+import eu.toolchain.async.AsyncFuture;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Metadata;
+import io.grpc.stub.MetadataUtils;
+import io.opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceRequest;
+import io.opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceResponse;
+import io.opentelemetry.proto.collector.metrics.v1.MetricsServiceGrpc;
+import io.opentelemetry.proto.common.v1.StringKeyValue;
+import io.opentelemetry.proto.metrics.v1.DoubleDataPoint;
+import io.opentelemetry.proto.metrics.v1.DoubleGauge;
+import io.opentelemetry.proto.metrics.v1.InstrumentationLibraryMetrics;
+import io.opentelemetry.proto.metrics.v1.ResourceMetrics;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OpenTelemetryPluginSink implements PluginSink {
+    private static final Logger log = LoggerFactory.getLogger(OpenTelemetryPluginSink.class);
+
+    @Inject private AsyncFramework async;
+    private String endpoint;
+    private Map<String, String> headers;
+    private ManagedChannel channel;
+    private MetricsServiceGrpc.MetricsServiceBlockingStub stub;
+
+    OpenTelemetryPluginSink(
+        String endpoint,
+        Map<String, String> headers
+    ) {
+        this.endpoint = endpoint;
+        this.headers = headers;
+    }
+
+
+    @Override
+    public void sendMetric(Metric metric) {
+        if (metric.hasDistribution()) {
+            log.warn("Distributions not supported, dropping metric");
+            return;
+        }
+
+        Map<String, String> tags = metric.getTags();
+
+        // NB(hexedpackets): Name is supposed to uniquely identify the metric. The combination
+        // of `key` and `what` may or may not be adequate. It might make sense to have this be
+        // configurable?
+        String name = metric.getKey() + "." + tags.get("what");
+        DoubleGauge gauge = convertGauge(metric);
+        io.opentelemetry.proto.metrics.v1.Metric.Builder metricBuilder =
+            io.opentelemetry.proto.metrics.v1.Metric.newBuilder()
+                .setName(name)
+                .setDoubleGauge(gauge);
+
+        String unit = tags.get("unit");
+        if (unit != null) {
+            metricBuilder.setUnit(unit);
+        }
+
+        ExportMetricsServiceRequest request = ExportMetricsServiceRequest.newBuilder()
```

Comment:
How is this metric used upstream ?

Reply:
Depends on the receiver. For Lightstep, the metrics are grouped on the name and then allow tag filtering/grouping under that.